src/sparseml/optim/manager.py

```diff
@@ -333,33 +401,71 @@ def iter_modifiers(self) -> Generator[None, None, BaseModifier]:
             for mod in modifiers_list:
                 yield mod
 
-    def to_string_lines(self) -> List[str]:
+    def to_string_lines(self, include_metadata: bool = False) -> List[str]:
```

Comment:
why default to false here and true above?

Reply:
The reason is --> if `def to_string_lines(self, include_metadata: bool = True)` this test would fail:
`tests/sparseml/pytorch/optim/test_manager.py::test_manager_staged_recipe_serialization`,
since it contains the following line of code:
`manager_yaml_str = str(manager)`.

What happens is: `str` calls dunder `__str__()` which in turn calls `to_string_lines` so that the recipe yaml gets assembled. Since this `manager` does not have any metadata AND we call `to_string_lines()` directly from `__str__()` and not from `save()`, this is how we prevent the test from failing.

So we either:
1. keep `include_metadata` as False to keep to old functionality
2. slightly change `tests/sparseml/pytorch/optim/test_manager.py::test_manager_staged_recipe_serialization` test.

I guess that the user is not expected to use `__str__()` but would rather assemble yaml through `save` function. This is why I have decided to make `include_metadata` default to False. 

Does it make sense? I am open for suggestions.